src/rectangle.ml

```diff
@@ -86,3 +86,38 @@ let empty =
   Array.Floatarray.unsafe_set arr 2 0.;
   Array.Floatarray.unsafe_set arr 3 0.;
   arr
+
+let get arr i = List.nth arr i
+
+let mindist p e =
+  let x0, y0, x1, y1 = (x0 e, y0 e, x1 e, y1 e) in
+  let gete ind = match ind with 0 -> x0 | 1 -> y0 | 2 -> x1 | _ -> y1 in
+  let dist i =
+    if get p i < gete i then (gete i -. get p i) *. (gete i -. get p i)
+    else if get p i > gete (i + 2) then
+      (get p i -. gete (i + 2)) *. (get p i -. gete (i + 2))
+    else 0.
+  in
+  dist 0 +. dist 1
+
+let minmaxdist p e =
+  let x0, y0, x1, y1 = (x0 e, y0 e, x1 e, y1 e) in
+  let gete ind = match ind with 0 -> x0 | 1 -> y0 | 2 -> x1 | _ -> y1 in
+  let rm k =
```

Comment:
A slightly more descriptive name would be helpful here, and also so we don't confuse `rm` and `rM`.